Libraries/LibIPC/MessageWindows.cpp

```diff
@@ -0,0 +1,60 @@
+/*
+ * Copyright (c) 2024, Tim Flynn <trflynn89@serenityos.org>
+ * Copyright (c) 2024, stasoid <stasoid@yahoo.com>
+ *
+ * SPDX-License-Identifier: BSD-2-Clause
+ */
+
+#include <LibIPC/Message.h>
+#include <io.h>
+
+namespace IPC {
+
+static void invalid_parameter_handler(wchar_t const*, wchar_t const*, wchar_t const*, unsigned int, uintptr_t)
+{
+}
+// Make _get_osfhandle return -1 instead of crashing on invalid fd in release (debug still __debugbreak's)
+static auto dummy = _set_invalid_parameter_handler(invalid_parameter_handler);
+
+using MessageSizeType = u32;
+
+MessageBuffer::MessageBuffer()
+{
+    m_data.resize(sizeof(MessageSizeType));
+}
+
+ErrorOr<void> MessageBuffer::extend_data_capacity(size_t capacity)
+{
+    TRY(m_data.try_ensure_capacity(m_data.size() + capacity));
+    return {};
+}
+
+ErrorOr<void> MessageBuffer::append_data(u8 const* values, size_t count)
+{
+    TRY(m_data.try_append(values, count));
+    return {};
+}
+
+ErrorOr<void> MessageBuffer::append_file_descriptor(int fd)
+{
+    intptr_t handle = _get_osfhandle(fd);
```

Comment:
Why not HANDLE? Parts of the code are using HANDLE parts intptr_t. Can this be made consistent?